### PR TITLE
feat(ContractEditor): refactor clausePlugin to use methods passed form props - 33

### PR DESCRIPTION
**Pull request by DianaLease**
https://github.com/accordproject/cicero-ui/issues/33

feat(ClausePlugin): load template using method passed via props
feat(ClausePlugin): parse clause using method passed via props

*FLAGS:*
- Corresponds with https://github.com/accordproject/template-studio-v2/pull/35

### CODE REVIEW

File: src/ContractEditor/index.js [hunk 1]
```diff
@@ -49,24 +52,30 @@ const contractProps = {
...
  */
-const ContractAssembler = props => (<div><MarkdownEditor markdown={props.markdown}
-        onChange={props.onChange}
-        lockText={true}
-        plugins={props.plugins}/></div>);
-
-const ContractEditor = () => ContractAssembler(contractProps);
+const ContractEditor = props => (
+  <MarkdownEditor
+    markdown={props.markdown || contractProps.markdown}
```
Comment (jeromesimeon) [inline src/ContractEditor/index.js]: Where is `contractProps` coming from?

Comment (DianaLease) [reply src/ContractEditor/index.js]: it's defined earlier in that same file (line 46)

File: src/ContractEditor/plugins/clausePlugin.js [hunk 2]
```diff
@@ -2,10 +2,6 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { Icon } from 'semantic-ui-react';
 import styled from 'styled-components';
-import { Template, Clause } from '@accordproject/cicero-core';
-import BadParse from '../../SlateCommands/BadParse';
```
Comment (jeromesimeon) [inline src/ContractEditor/plugins/clausePlugin.js]: Should the `BadParse` `GoodParse` `SetNodeData` be removed as well? Or are those called from the editor instead?

File: src/ContractEditor/index.js (hunk 1 above)
Comment (jeromesimeon) [inline src/ContractEditor/index.js]: Where is `contractProps` coming from?

File: src/ContractEditor/plugins/clausePlugin.js (hunk 2 above)
Comment (DianaLease) [reply src/ContractEditor/plugins/clausePlugin.js]: as far as I can see, they are not being used anywhere else. I'll remove those files, thanks! We'll be implementing new functions to add annotations once Slate pushes a fix to that bug that Dan was experiencing.

File: src/ContractEditor/index.js (hunk 1 above)
Comment (DianaLease) [reply src/ContractEditor/index.js]: it is defined on line 46

Comment (jeromesimeon) [reply src/ContractEditor/index.js]: Right. I had missed that. Thanks!

File: src/ContractEditor/plugins/clausePlugin.js [hunk 3]
```diff
@@ -54,29 +50,16 @@ function ClausePlugin() {
...
-      }
-    } else {
-      console.log('Template not found on node.');
-    }
-
+    const nodeAttributes = clauseNode.data.get('attributes');
+    const { src, clauseid } = nodeAttributes;
 
+    parseClause(src, textNode.text.trim(), clauseid)
+      .then(parseResult => console.log(parseResult)) // add/remove annotation
```
Comment (irmerk) [inline src/ContractEditor/plugins/clausePlugin.js]: Not sure I entirely understand what `parseClause` is doing here, do we want it to be logging the result to the console?

Comment (DianaLease) [reply src/ContractEditor/plugins/clausePlugin.js]: The comment is what we actually want in the future (to add or remove an annotation via slate). We'll be implementing new functions to add annotations once Slate pushes a fix to that bug that Dan was experiencing.

**Pull request merged**